web/packages/shared/components/UnifiedResources/UnifiedResources.tsx

```diff
@@ -124,6 +128,17 @@ interface UnifiedResourcesProps {
   availableKinds: SharedUnifiedResource['resource']['kind'][];
   setParams(params: UnifiedResourcesQueryParams): void;
   onLabelClick(label: ResourceLabel): void;
+  // bulkActions allow us to pass in any features we want from the parent
+  // that interact with selected items (example: add all selected resources to an access request)
+  // TODO (avatus): pinning should be removed from the internals of UnifiedResources as passed in
+  // from the parents.
+  bulkActions: {
+    // key is an arbitrary name of what the bulk action is, as well
+    // as the key used when mapping our action components
+    key: string;
+    Component: (onClick: () => void) => React.ReactElement;
```

Comment:
I know that I proposed it, but I think that there is no need to pass a React component here. 
All the items will probably look the same, so why not have only `Icon`, `name` and `action`? We will have more control on what is passed to the component (and it will be just a bit simpler).

Reply:
I think we'd need the component so we can add extra logic when needed, such as pinning support (with the hover wrapper and all that)